src/RedisResourceManager.php

```diff
@@ -315,7 +361,9 @@ protected function connect(array &$resource)
             }
 
             $resource['initialized'] = true;
-            if ($resource['password']) {
+            if ($resource['user'] && $resource['password']) {
+                $redis->auth([$resource['user'], $resource['password']]);
```

Comment:
I checked the phpredis [`redis.stub.php`](https://github.com/phpredis/phpredis/blob/62cf943fecc5182c6329b332df43cf28012cef55/redis.stub.php#L586C1-L587C1) and if I understand it correct, the value has to be an array:
```php
$redis->auth(['username', 'password']);
```

So I guess we have to adapt here and also fix the unit test.